code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1190,3 +1190,24 @@
   */
 /mob/living/carbon/proc/get_biological_state()
 	return BIO_FLESH_BONE
+
+/// Returns whether or not the carbon should be able to be shocked
+/mob/living/carbon/proc/should_electrocute(power_source)
+	if (ismecha(loc))
+		return FALSE
+
+	if (wearing_shock_proof_gloves())
+		return FALSE
+
+	if(!get_powernet_info_from_source(power_source))
+		return FALSE
+
+	return TRUE
+
+/// Returns if the carbon is wearing shock proof gloves
+/mob/living/carbon/proc/wearing_shock_proof_gloves()
+	if (gloves)
+		var/obj/item/clothing/gloves/G = gloves
+		return G.siemens_coefficient == 0
+
+	return FALSE
```

Comment:
Do you take into account things such as Genetics Insulated mutation and Nerve Grounding surgery as part of any of these checks?

Reply:
These are more or less copied and pasted from the old checks, so they were never able to. I can potentially add these in though.

Reply:
Yes, please!

Reply:
Don't scope creep features into a fix pr, rule number 1

Reply:
Read the code before you comment, rule number 0.

/datum/wires/airlock/interactable(mob/user) returns FALSE early if the airlock is electrified and the user's should_electrocute proc returns TRUE. This part is fine.

should_electrocute() only checks for insuls when it comes to sources of shock immunity. It doesn't check for any other type of shock immunity and will return TRUE despite a player being shock immune.

As a result, people who are immune through other sources cannot interact with an electrified door when this proc feeds its return value back into interactable(...).

This is a bug and should ideally be fixed before the PR is merged, otherwise we're gonna have to wait another 24+ hours after it's merged for some poor schmuck like me to fix it after the fact.

This bug has been introduced because the old check for interactibility was `if(!issilicon(user) && A.isElectrified() && A.shock(user, 100))` - A.shock would return FALSE when the person had some form of TRAIT shock immunity (such as from surgery or genetics)

It goes through a chain of calls where a person with one of the immunities granted by traits would take 0 damage, 0 power would be drained from some power cell and an electrocute_mob proc would return 0 for the total energy drained from the cell. This FALSEy value fed through A.shock() causing it to return FALSE and thus the `if(!issilicon(user) && A.isElectrified() && A.shock(user, 100))` check would resolve to FALSE and interactable() would not return early.

In essence, A.shock() was indirectly checking for the appropriate immunities. This functionality is missing in the above code as should_electrocute doesn't do the relevant checks.

tl;dr - A.shock() eventually led to /mob/living/proc/electrocute_act() which would cause a chain of FALSEy returns which eventually led to A.shock() returning FALSE when a player had the appropriate immunities.

should_electrocute() doesn't replicate these checks. As a result, people who are shock immune are prevented from interacting with shocked doors, which is a bug compared to live code where people who are shock immune are able to interact with shocked doors.

oranges should probably read the code more before commenting.

Reply:
There's no need to be rude, your original message made no such detailed claims.

Had you been clearer in communicating originally, there would be no room for misinterpretation.

Reply:
Apologies for the hostility and rudeness. I think you caught me at a grumpy time.